### PR TITLE
Update default name for account and watched address #18416

### LINKED ISSUE #18416: Wallet - Update default name for watched address when adding it
# Feature Issue

When adding a new watched address, the placeholder should  be `Watched address (x)` where `x` is the total number of watched addresses in the profile + 1.

### Design

[Figma Link](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=2052%3A522184&mode=design&t=H2vraH5nSHXL10D8-1)

### Acceptance Criteria

- The placeholder name for the watched address should display `Watched address (x)` 
- The placeholder name for a generated account should display `Account (y)` where `y` is the total number of accounts (not watched) in the profile + 1.
- the actual name should be empty and the user has to input some value themselves

### PR DESCRIPTION
**Pull request by Rende11**
Fixes #18416

### Summary
Upd:
This PR updates **placeholders** for account and watched address when creating.

### Review and testing notes
* autofocus - sometimes it doesn't shows
* disabled sliding button while text-input is empty - not implemented due to issues related to this functionality in the component (you can't slide button after changing `disabled?` state)

### Before and after comparison

https://github.com/status-im/status-mobile/assets/10757633/e46f634c-0404-4748-81be-50dc2aeb4ec0

status: ready

### CODE REVIEW

**Comment by J-Son89** [comment]
seems the component tests are failing btw
Screenshot 2024-01-31 at 16 34 51

File: src/status_im/contexts/wallet/add_address_to_watch/confirm_address/view.cljs [hunk 1]
```diff
@@ -15,8 +15,8 @@
 (defn view
   []
   (let [{:keys [address]}  (rf/sub [:get-screen-params])
-        number-of-accounts (count (rf/sub [:profile/wallet-accounts]))
-        account-name       (reagent/atom (i18n/label :t/default-account-name
+        number-of-accounts (count (filter :watch-only? (rf/sub [:wallet/accounts])))
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/add_address_to_watch/confirm_address/view.cljs]: @J-Son89 should we have subs that returns watch-only accounts like `(rf/sub [:wallet/watch-only-accounts])` and active accounts like `(rf/sub [:wallet/active-accounts])` (or something similar)

Comment (J-Son89) [reply src/status_im/contexts/wallet/add_address_to_watch/confirm_address/view.cljs]: Yeah makes sense. Can you add these subs here @Rende11 as it’s related to this pr?

**Review by J-Son89 — APPROVED** [review]

File: src/status_im/contexts/wallet/add_address_to_watch/confirm_address/view.cljs (hunk 1 above)
Comment (Rende11) [reply src/status_im/contexts/wallet/add_address_to_watch/confirm_address/view.cljs]: @J-Son89 Sure

Comment (Rende11) [reply src/status_im/contexts/wallet/add_address_to_watch/confirm_address/view.cljs]: Done

**Review by OmarBasem — APPROVED** [review]

File: src/status_im/subs/wallet/wallet.cljs [hunk 2]
```diff
@@ -95,6 +95,18 @@
         (map #(assoc-network-preferences-names network-details % test-networks-enabled?))
         (sort-by :position))))
 
+(rf/reg-sub
+ :wallet/active-accounts
+ :<- [:wallet/accounts]
+ (fn [accounts]
+   (remove :watch-only? accounts)))
```
Comment (smohamedjavid) [inline src/status_im/subs/wallet/wallet.cljs]: @Rende11 - We have an existing sub `wallet/accounts-without-watched-accounts` which does the same. Can we use it?

Comment (Rende11) [reply src/status_im/subs/wallet/wallet.cljs]: Done

**Review by smohamedjavid — APPROVED** [review]
Nice work! 🙌

**Comment by Rende11** [comment]
https://github.com/status-im/status-mobile/assets/10757633/5163283a-e958-4356-8bfb-0e79110149ca

**Comment by Rende11** [comment]
You can't slide button after changing `disabled?` state


https://github.com/status-im/status-mobile/assets/10757633/b2a001f3-e21b-4c93-8139-0cb71888784e

**Comment by J-Son89** [comment]
@status-im/mobile-qa - when testing this feature, please note some details. We discussed with designers and it should be the default placeholder with this naming convention here. The user is expected to enter a name, for that reason the slide button should be disabled when the name is empty, however there is an issue with disabling the slide button so @Rende11 did not disable it here. 
We will fix the slide button in a separate issue and then address this too. 👍

**Comment by mariia-skrypnyk** [comment]
Hi @Rende11 !

Thanks for your PR! 
I've checked placeholders for account and watched address when creating and it works great.

**ISSUE 1:** 

The only thing from my side is to ask design team regarding font color as I can see [here](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=2052-522184&mode=design&t=cptGlntgDtVpa8Tb-0) not the color used by you (for both: accounts and watch only)
Screenshot 2024-02-15 at 12 06 52
Please, @Francesca-G let me know if it is a correct implementation?

**Review by Francesca-G — APPROVED** [review]
@mariia-skrypnyk you're correct, the color of the placeholder doesn't match design. 
I can also confirm the slider issue @J-Son89 mentioned, as well as other issues with it, as commented [here](https://www.figma.com/file/FuHW0rmIa5fQr5yyt3pXM6/Review-Wallet-%2F-Mobile?type=design&node-id=13%3A941&mode=design&t=mUDlGTo8Wounh6eJ-1)

Adding follow up required 👍

**Comment by J-Son89** [comment]
Thanks @Francesca-G, @mariia-skrypnyk if it's okay we will handle these issues separately 👍

**Comment by mariia-skrypnyk** [comment]
ISSUE 1 [followup](https://github.com/status-im/status-mobile/issues/18856)

**Comment by J-Son89** [comment]
@mariia-skrypnyk - here is the issue for disabling the slide button - https://github.com/status-im/status-mobile/issues/18894

**Pull request merged**